test/test_sync.cpp

```diff
@@ -770,7 +770,7 @@ TEST(Sync_DetectSchemaMismatch_ColumnType)
             REALM_ASSERT(error_info);
             std::error_code ec = error_info->error_code;
             bool is_fatal = error_info->is_fatal();
-            CHECK(ec == sync::Client::Error::bad_changeset || ec == sync::ProtocolError::invalid_schema_change);
+            CHECK(ec == sync::Client::Error::bad_changeset || ec == sync::ProtocolError::bad_changeset);
```

Comment:
good catch. it is not needed, I must have changed them due to some previous implementation.

Reply:
On a second look, the C++ sync server never sends an `invalid_schema_change` error. The reason why there could be 2 errors is because it depends which side detects problems with the schema first. It seems that before the client always detected it first (i.e, bad download) and closed the connection, whereas now, sometimes, on some platforms, the server sends an error first as a result of integrating a bad changeset itself (bad upload).

Below, it can be clearly seen that the server sends bad_changeset(212) as opposed to  `invalid_schema_change`(225).
`Server: Sync Connection[2]: Session[1]: Sending: ERROR(error_code=212, message_size=220, try_again=false)`

Reply:
This was in the end handled in `receive_error_message` by ignoring errors from the server if the client already discovered an error.